File: benchmarks/competitors/calls.c
#include <stdio.h>

static long long add(long long a, long long b) {
    return a + b;
}

int main(void) {
    long long total = 0;
    for (long long i = 0; i < 1000000; i++) {
        total = add(total, 1);
    }
    printf("%lld\n", total);
    return 0;
}

File: benchmarks/competitors/calls.rs
#[inline(never)]
fn add(a: i64, b: i64) -> i64 {
    a + b
}

fn main() {
    let mut total: i64 = 0;
    let mut i: i64 = 0;
    while i < 1_000_000 {
        total = add(total, 1);
        i += 1;
    }
    println!("{total}");
}
